examples/applications/plot_time_series_lagged_features.py

```diff
@@ -1,425 +1,435 @@
-        ax=ax,
-    )
-    ax.set(xlabel="Predicted demand", ylabel="True demand")
-    ax.legend(["Best model", label])
-
-plt.show()
-
-# %%
-# Conclusion
-# ----------
-# Through this example we explored time series forecasting using lagged
-# features. We compared a naive regression (using the standardized
-# :class:`~sklearn.model_selection.train_test_split`) with a proper time
-# series evaluation strategy using
-# :class:`~sklearn.model_selection.TimeSeriesSplit`. We observed that the
-# model trained using :class:`~sklearn.model_selection.train_test_split`,
-# having a default value of `shuffle` set to `True` produced an overly
-# optimistic Mean Average Percentage Error (MAPE). The results
-# produced from the time-based split better represent the performance
-# of our time-series regression model. We also analyzed the predictive uncertainty
-# of our model via Quantile Regression. Predictions based on the 5th and
-# 95th percentile using `loss="quantile"` provide us with a quantitative estimate
-# of the uncertainty of the forecasts made by our time series regression model.
-# Uncertainty estimation can also be performed
-# using `MAPIE <https://mapie.readthedocs.io/en/latest/index.html>`_,
-# that provides an implementation based on recent work on conformal prediction
-# methods and estimates both aleatoric and epistemic uncertainty at the same time.
-# Furthermore, functionalities provided
-# by `sktime <https://www.sktime.net/en/latest/users.html>`_
-# can be used to extend scikit-learn estimators by making use of recursive time
-# series forecasting, that enables dynamic predictions of future values.
+"""
+===========================================
+Lagged features for time series forecasting
+===========================================
+
+This example demonstrates how Polars-engineered lagged features can be used
+for time series forecasting with
+:class:`~sklearn.ensemble.HistGradientBoostingRegressor` on the Bike Sharing
+Demand dataset.
+
+See the example on
+:ref:`sphx_glr_auto_examples_applications_plot_cyclical_feature_engineering.py`
+for some data exploration on this dataset and a demo on periodic feature
+engineering.
+
+"""
+
+# %%
+# Analyzing the Bike Sharing Demand dataset
+# -----------------------------------------
+#
+# We start by loading the data from the OpenML repository as a raw parquet file
+# to illustrate how to work with an arbitrary parquet file instead of hiding this
+# step in a convenience tool such as `sklearn.datasets.fetch_openml`.
+#
+# The URL of the parquet file can be found in the JSON description of the
+# Bike Sharing Demand dataset with id 44063 on openml.org
+# (https://openml.org/search?type=data&status=active&id=44063).
+#
+# The `sha256` hash of the file is also provided to ensure the integrity of the
+# downloaded file.
+import numpy as np
+import polars as pl
+
+from sklearn.datasets import fetch_file
+
+pl.Config.set_fmt_str_lengths(20)
+
+bike_sharing_data_file = fetch_file(
```

Comment:
Two questions about this:
- should we use `sha256` argument? What's the worst that can happen if we don't use it? Using a corrupted file and have a weird error? Using it in this example kind of makes `fetch_file` look not super convenient to use. Originally I thought by looking at the code in this example that the `sha256` was required but it's not.
- I am guessing the URL is likely to change (my understanding is that this is some kind of OpenML internal details), but we can wait and see if that actually happens.

Reply:
OK fair enough, let's leave it like this then!

On weird thing in particular, I think, is that there is no way to know the `sha256sum` in advance in the OpenML parquet case, but maybe one day there will be one.